support/Python/DirectoryStructure.py

```diff
@@ -126,3 +126,99 @@ def list_segments(segments_dir: Union[str, Path]) -> List[Segment]:
         return []
     matches = map(Segment.match, segments_dir.iterdir())
     return sorted(match for match in matches if match)
+
+
+@dataclass(frozen=True, order=True)
+class PipelineStep:
+    """A complete simulation iteration
+
+    Represents a full simulation run. Multiple iterations can exist,
+    organized in a directory structure like this:
+
+    ```
+    BASE_DIR/
+        001_InitialData/
+            Segments/
+                Segment_0000/
+                    Checkpoints/
+                    InputFile.yaml
+                    Submit.sh
+                Segment_0001/
+                    ...
+        002_Inspiral/
+            ...
+    ```
+
+    Note: "InitialData" and "Inspiral" are examples, any name can be used.
+    """
+
+    path: Path
+    id: int
+    label: str
+
+    NAME_PATTERN = re.compile(r"(\d+)_(.+)")
+    NUM_DIGITS = 3
+
+    @classmethod
+    def first(cls, directory: Path, label: str) -> "PipelineStep":
+        """Create the first directory in a sequence"""
+        name = "0" * cls.NUM_DIGITS + "_" + label
+        return PipelineStep(path=directory / name, id=0, label=label)
+
+    # @classmethod
+    # def next(self) -> "PipelineStep":
+    #     """Get the next directory in the sequence"""
+    #     next_id = self.id + 1
+    #     next_name = "PipelineStep_" + str(next_id).zfill(self.NUM_DIGITS)
+    #     return PipelineStep(
+    #         path=self.path.resolve().parent / next_name, id=next_id
+    #     )
+    def next(self, label: Optional[str] = None) -> "PipelineStep":
+        """Get the next directory in the sequence"""
+        next_id = self.id + 1
+        next_label = label if label else self.label
```

Comment:
You can delete this line because `label` should be required (I don't think the next pipeline step should have the previous label by default)